### PR TITLE
Set 204 HTTP response + CORS on empty MVT tiles (see #278)

### LINKED ISSUE #278: Vector tiles : 404 has no CORS headers in response
Hello,

We use with Mapbox GL the Osmose tiles on projetdumois.fr. When a vector tile has no data, current code send a HTTP 404 status. But the response doesn't contain any CORS header, making client span a lot of errors :

Capture d’écran_2020-09-29_09-08-59

Example request : https://osmose.openstreetmap.fr/api/0.3/issues/10/504/352.mvt?item=8370&country=france*

HTTP response headers :

```
HTTP/2 404 Not Found
server: nginx/1.19.1
date: Tue, 29 Sep 2020 07:09:25 GMT
content-type: text/plain
content-length: 0
X-Firefox-Spdy: h2
```

Would it be possible either :
- To send a valid tile (200) status but with an empty list of features
- Or to add CORS headers in HTTP response ?

Best regards.

### PR DESCRIPTION
**Pull request by PanierAvide**
To fix #278

### CODE REVIEW

File: api/issues_tiles.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def issues_mvt(db, z, x, y, format):
             response.content_type = 'application/vnd.mapbox-vector-tile'
             return tile
         else:
-            return HTTPError(404)
+            return HTTPResponse(status=204, headers={'Access-Control-Allow-Origin': '*'})
```
Comment (frodrigo) [inline api/issues_tiles.py:194]: The CORS is added globally, not to each request.
https://github.com/osm-fr/osmose-frontend/blob/master/modules/bottle_cors.py

Comment (frodrigo) [inline api/issues_tiles.py:194]: Should do the job.
```suggestion
            return HTTPResponse(status=204)
```

Comment (PanierAvide) [reply api/issues_tiles.py:194]: That was what I thought, but in fact it doesn't (tried with local frontend setup) : if you use HTTPResponse, CORS headers aren't added automatically. If using HTTPResponse, you have either to set this explicitly, or use some module for Bottle (?). Source for this : https://stackoverflow.com/questions/53640565/returning-httpresponse-object-in-bottle-loses-response-headers

Comment (frodrigo) [reply api/issues_tiles.py:194]: Ok.

**Comment by frodrigo** [comment]
Deployed

**Comment by PanierAvide** [comment]
Thanks :smiley:

**Pull request merged**